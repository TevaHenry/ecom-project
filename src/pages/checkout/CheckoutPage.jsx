import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectCartItems, selectCartTotal } from '../../redux/Cart/cartSelectors';

import CheckoutItem from '../../components/CheckoutItem/CheckoutItem';
import StripeCheckoutButton from '../../components/StripeButton/StripeButton';
import {
  CheckoutPageContainer, 
  CheckoutHeaderContainer,
  HeaderBlockContainer,
  TestWarningContainer, 
  CheckoutTotalContainer
} from './CheckoutPageStyles';

const CheckoutPage = ({ cartItems, cartTotal }) => (
  <CheckoutPageContainer>
    <CheckoutHeaderContainer>
      <HeaderBlockContainer><span>Product</span></HeaderBlockContainer>
      <HeaderBlockContainer><span>Descriptioon</span></HeaderBlockContainer>
      <HeaderBlockContainer><span>Quantity</span></HeaderBlockContainer>
      <HeaderBlockContainer><span>Price</span></HeaderBlockContainer>
      <HeaderBlockContainer><span>Remove</span></HeaderBlockContainer>
    </CheckoutHeaderContainer>
    {cartItems.map(cartItem => <CheckoutItem key={cartItem.id} cartItem={cartItem} />)}
    <CheckoutTotalContainer>Total: ${cartTotal}</CheckoutTotalContainer>
    <TestWarningContainer>
      *Please use the following test credit card for payments*
      <br/>
      4242 4242 4242 4242 - Exp: 01/20 - CVV: 123
    </TestWarningContainer>
    <StripeCheckoutButton price={cartTotal} />
  </CheckoutPageContainer>
)

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  cartTotal: selectCartTotal
})

export default connect(mapStateToProps)(CheckoutPage);