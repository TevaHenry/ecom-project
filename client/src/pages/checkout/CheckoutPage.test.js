import { shallow } from 'enzyme';
import React from 'react';
import { CheckoutPage } from './CheckoutPage';

let wrapper;
beforeEach(() => {
  const mockProps = {
    cartItems: [],
    total: 100
  };

  wrapper = shallow(<CheckoutPage {...mockProps} />);
});

it('expect to render CheckoutPage component', () => {
  expect(wrapper).toMatchSnapshot()
})