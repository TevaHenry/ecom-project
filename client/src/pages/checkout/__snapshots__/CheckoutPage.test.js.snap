// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CheckoutPage component 1`] = `ShallowWrapper {}`;
